cdap-security/src/main/java/co/cask/cdap/security/server/JASPIAuthenticationHandler.java

```diff
@@ -114,9 +113,8 @@ protected Configuration getLoginModuleConfiguration() {
         String configRegex = Constants.Security.AUTH_HANDLER_CONFIG_BASE.replace(".", "\\.").concat(".");
         HashMap<String, String> configurables = (HashMap<String, String>) configuration.getValByRegex(configRegex);
 
-        Iterator it = configurables.entrySet().iterator();
-        while (it.hasNext()) {
-          Map.Entry pairs = (Map.Entry) it.next();
+        for (Object o : configurables.entrySet()) {
```

Comment:
ehm. this code needs to be rewritten. getByValRegex already returns a Map<String, String>